File: 이진영/BruteForce/피로도.js
function solution(k, dungeons) {    
    let max = 0;
    let visited = Array(dungeons.length).fill(false);
    
    const getCount = (k, dungeons, visited, count) => {
        if (visited.every(x => x)) {
            max = Math.max(max, count);
            return;
        }
        
        for (let i = 0; i < dungeons.length; i++) {
            if (visited[i]) continue;
            
            visited[i] = true;
            getCount(k, dungeons, visited, count);
            if (k >= dungeons[i][0])
                getCount(k - dungeons[i][1], dungeons, visited, count + 1);
            visited[i] = false;
        }
    }
    
    getCount(k, dungeons, visited, 0);
    return max;
}

File: 김채린/BruteForce/피로도.kt
class Solution {
    var maxCount = -1

fun solution(
    k: Int,
    dungeons: Array<IntArray>,
): Int {
    var total = 0
    var visited = Array<Boolean>(dungeons.size) { false }
    back(visited, 0, k, dungeons)
    return maxCount
}

fun back(
    visited: Array<Boolean>,
    count: Int,
    now: Int,
    dungeons: Array<IntArray>,
) {
    if (now < 1) {
        maxCount = Math.max(count, maxCount)
        return
    }
    for (i in 0 until visited.size) {
        if (!visited[i] && now >= dungeons[i][0]) {
            visited[i] = true
            back(visited, count + 1, now - dungeons[i][1], dungeons)
            visited[i] = false
        }
    }
    maxCount = Math.max(count, maxCount)
}
}
